Upgrade EUI to v95.10.1

cee-chen opened the pull request:
`v95.9.0`⏩`v95.10.1`

> [!note]
> **EuiDataGrid**'s header cells have received a major UX change in order to support interactive children within header content. Column header actions now must be hovered and then clicked directly, or opened with the Enter key, as opposed to being able to click the entire header cell to see the actions popover.

_[Questions? Please see our Kibana upgrade FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)_

---

## [`v95.10.0`](https://github.com/elastic/eui/releases/v95.10.0)

- Updated `EuiDataGrid` to support interactive header cell content ([#7898](https://github.com/elastic/eui/pull/7898))
- Updated `EuiSearchBar`'s `field_value_selection` filter type with a new `autoSortOptions` config, allowing consumers to configure whether or not selected options are automatically sorted to the top of the filter list ([#7958](https://github.com/elastic/eui/pull/7958))
- Updated `getDefaultEuiMarkdownPlugins` to support the following new default plugin configurations: ([#7985](https://github.com/elastic/eui/pull/7985))
  - `parsingConfig.linkValidator`, which allows configuring `allowRelative` and `allowProtocols`
  - `parsingConfig.emoji`, which allows configuring emoticon parsing
  - `processingConfig.linkProps`, which allows configuring rendered links with any props that `EuiLink` accepts
  - See our **Markdown plugins** documentation for example `EuiMarkdownFormat` and `EuiMarkdownEditor` usage
- Updated `EuiDatePicker` to support `append` and `prepend` nodes in its form control layout ([#7987](https://github.com/elastic/eui/pull/7987))

**Bug fixes**

- Fixed border rendering bug with inline `EuiDatePicker`s with `shadow={false}` ([#7987](https://github.com/elastic/eui/pull/7987))
- Fixed `EuiSuperSelect`'s placeholder text color to match other form controls ([#7995](https://github.com/elastic/eui/pull/7995))

**Accessibility**

- Improved the keyboard navigation and screen reader output for `EuiDataGrid` header cells ([#7898](https://github.com/elastic/eui/pull/7898))

## [`v95.10.1`](https://github.com/elastic/eui/releases/v95.10.1)

**Bug fixes**

- Fixed a visual bug in compact density `EuiDataGrid`s, where the header cell height would increase when the actions button became visible ([#7999](https://github.com/elastic/eui/pull/7999))

cee-chen committed:
Upgrade EUI to v95.10.0

cee-chen committed:
i18n updates

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

cee-chen commented:
/ci

cee-chen committed:
Fix failing kbn-unified-data-table tests

cee-chen committed:
Fix several test console errors/warnings

- not all of them, just the oens related to header column actions in this PR. someone on the owning Kibana team should likely look into resolving all of them

- errors were `Warning: An update to DataTable inside a test was not wrapped in act(...).` and `Warning: Can't perform a React state update on an unmounted component`

cee-chen committed:
Fix other Jest tests toggling column actions to use data-test-subj instead of text

cee-chen committed:
[FTR] Fix tests to account for DOM changes

cee-chen committed:
[FTR] Fix EuiDataGrid service to account for interactive headers/actions button changes

+ fix misc failing test to use said service

cee-chen committed:
[FTR] Fix failing Observability tests to use dataGrid service that accounts for header changes

+ add new util for default "Hide columns" copy

cee-chen committed:
[FTR] Fix failing tests to with new dataGrid service that allows clicking any action at an index

+ add new util for default "Hide columns" copy

mgadewoll committed:
[FTR]: update getHeaderFields selector

mgadewoll committed:
[FTR]: update lense page changeTableSortingBy hover position to ensure mouse is inside header column

mgadewoll committed:
[FTR]: update DataGrid service selectors

mgadewoll committed:
chore: cleanup unsed var

mgadewoll committed:
[FTR]: revert table reference in DataGrid service getHeaderFields

ensures test order instead, as opened flyout contains a standalone datagrid the column header test assertion should happen on a state with only the first datagrid

mgadewoll committed:
[FTR]: update test order

cee-chen committed:
Patch v95.10.1

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui/v95.10.0

cee-chen commented on test/functional/services/data_grid.ts:
```diff
@@ -427,15 +427,27 @@ export class DataGridService extends FtrService {
       await this.openColMenuByField(field);
     } else {
+      const columns = await this.find.allByCssSelector(
+        `[data-test-subj^="dataGridHeaderCell-"]:not(.euiDataGridHeaderCell--controlColumn)`
```
```suggestion
        `.euiDataGridHeaderCell:not(.euiDataGridHeaderCell--controlColumn)`
```

cee-chen committed:
Simplify test selector

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

kdelemme reviewed (approved):
obs-ux-management code change looks good to me!

markov00 reviewed (approved):
Code review and visual check, good for me

cee-chen committed:
Merge branch 'main' into eui/v95.10.0

davismcphee reviewed (approved):
Data Discovery changes look good, and everything seems to be working as expected in Discover 👍 Thanks for making the data grid header change, this unblocks some planned changes for us!

cee-chen committed:
Merge branch 'main' into eui/v95.10.0

cee-chen commented:
@elastic/kibana-core @elastic/security-threat-hunting-investigations Ping for codeowning review - we will be requesting an admin merge by today or tomorrow.

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui/v95.10.0

cee-chen committed:
fix type failure
